### PR TITLE
AsteroidStation, a slightly larger alternative to Box

### PR DESCRIPTION
**Pull request by MenacingManatee**
# Document the changes in your pull request

This concept was originally inspired as an alternative to box and icemoon that provided more to do while still on the station z level, with the idea that this is a hidden station within an asteroid belt nearby lavaland. When I saw kilo was doing something similar I started taking inspiration from that as well. 

Asteroid Station is roughly the size of box, and is made with a population of 25-80 in mind. The mining areas and hidden ruins therein should provide extra space and activities in higher pops, while not causing the station to feel too large or empty on the lower end.

The station consists of 1 large asteroid containing the main station, 2 medium sized asteroids containing the AI satellite and a bunker of russians hired by the syndicate to spy on the station, and 3 small asteroids, one of which contains a random maint ruin. The right side of the station, with the bottom right in particular, is meant to have a unfinished feel as if construction was started but never finished, with maintenance tunnels that are longer and more winding, a few dead ends, and boarded off maint rooms leading to the mining area. The mining area also contains a few hidden ruins that can be discovered in a round.

The station is built around the research department at the center, with engineering in the bottom right, medbay below research, charge to the bottom left, command to the left, service across the station above research, bisected by a hall leading to security above that. The AI satellite is relatively far from the station, with the two main access points being teleport pads from the bridge and from the RD's office, with an addition access point being catwalks from engineering to connect power and atmos. The station only has 3 solar array compared to box's 4, with one of them being extra large to compensate.

TODO:
- [x] Disposals
- [x] Light switches, fire alarms, extinguishers, air alarms
- [x] AI holopads
- [x] Cameras
- [x] Double check all doors have correct access and names
- [x] Signage, paintings, posters
- [x] Test power and atmos to check for misplaced/missing connections
- [x] Add more windows decorations to hallways to make them feel less empty
- [x] Firelocks
- [x] Intercomms
- [x] Bounty boards, newscasters, & requests consoles

# Wiki Documentation

<details>
<summary>Old Version</summary

AsteroidStation-1

</details>

<details>
<summary>New Version</summary>

image

</details>

# Changelog

:cl:  
rscadd: Adds a brand new station built into an asteroid belt - Currently in Alpha Testing
/:cl:

### CODE REVIEW

**Comment by DukeBobRoss** [comment]
This shouldn't be merged until wormholes can no longer teleport you into rocks, otherwise anytime wormholes hit like 6 people are gonna be removed from the round permanently.

**Comment by ToasterBiome** [comment]
I see you replaced the minerals with ones that have air in them, but the floors still do not.

image
pipes through wall

image

I don't like the wires through the maint spawners

image

airlock controller + pipes

needs firelocks everywhere

image
plating

image

this whole hallway is weird and makes cargo weird to go to :(

missing some maint doors, also maint door helpers

image


pipes through window, also plating under

image


trim around pool


ill look at more later

**Comment by N3D6** [comment]
This is cool as hell, map variety is based

**Comment by MenacingManatee** [comment]
> I see you replaced the minerals with ones that have air in them, but the floors still do not.
>
> pipes through wall
> 
> I don't like the wires through the maint spawners
> 
> airlock controller + pipes
> 
> needs firelocks everywhere
> 
> this whole hallway is weird and makes cargo weird to go to :(
> 
> missing some maint doors, also maint door helpers
> 
> pipes through window, also plating under
> 
> trim around pool
> 
> ill look at more later

Floors have air
image

Fixed pipes through walls/window
image
image

Fixed the wires
image

Pool has trim now
image

Plating 
image

Controller+pipes
image

I'll have to think on cargo, and firelocks are on my todo list

**Comment by Anvilman6** [comment]
This is pretty damn amazing Manatee, massive kudos to you for actually getting this put together! There's a few places I think could be touched up (which I'll probably just message you directly about), but for the most part you've done a bang up job.

I will say here, however, that the Russian asteroid kind of worries me. It seems like that'd be a prime locale for bored miners (or the Curator with their space suit) to hit every round for whatever goodies it may hold if it's a static spawn.

**Comment by N3D6** [comment]
> This shouldn't be merged until wormholes can no longer teleport you into rocks, otherwise anytime wormholes hit like 6 people are gonna be removed from the round permanently.

Skill issue really. It’s no different than being teleported into space, it just requires a little more effort to get the body. If you get teleported into rocks without yelling about it and having your suit sensors on, if you don’t get saved it’s your own fault.

Any paramedic with a tracking device and a pka or a pickaxe can find you very quickly in this scenario

**Comment by MenacingManatee** [comment]
> I will say here, however, that the Russian asteroid kind of worries me. It seems like that'd be a prime locale for bored miners (or the Curator with their space suit) to hit every round for whatever goodies it may hold if it's a static spawn.

There are a few goodies, but nothing that should be majorly problematic for security, especially considering the difficulty in getting it. The most concerning thing you can get there is a single gun from a mosin russian, which I'm considering swapping out for another melee russian

**Comment by ToasterBiome** [comment]
don't forget to look at map_errors

**Comment by MenacingManatee** [comment]
Map post updates
AsteroidStation-1

**Comment by ShatteredParadox** [comment]
A good alternative. Interesting choice to limit security's connection to the main station. Makes breaking in difficult to do sneakily. Im not a huge fan of the circulation pathway that takes prisoners through a secure area to perma brig, makes perma feel a little misplaced. There are also a few regular walls separating perma from security officer locker room that would probably benefit from being made reinforced.

**Comment by MenacingManatee** [comment]
> There are also a few regular walls separating perma from security officer locker room that would probably benefit from being made reinforced.

Good catch, fixed

**Comment by Boxlta** [comment]
The incinerator shouldnt be next to a major hallway or the radiation from any fusion done will leak over into the evac hall

**Comment by MenacingManatee** [comment]
> The incinerator shouldnt be next to a major hallway or the radiation from any fusion done will leak over into the evac hall

Fixed
image

**Comment by JamieD1** [comment]
Under Review

**Comment by JamieD1** [comment]
Sorry, this will be on the backburner for a little bit. Hopefully not too long though.

**Comment by Moja10** [comment]
image
MANATEE ROOM YOGGERS

**Comment by JamieD1** [comment]
Feedback 2 Given

**Comment by JamieD1** [comment]
Ongoing

**Comment by adamsong** [comment]
What is the status on this?

**Comment by MenacingManatee** [comment]
Still in progress, I made a major shift to the layout and positioning of cargo and command, trying to smooth it out in my free time still

**Comment by MenacingManatee** [comment]
No clue why it auto requested reviews, I just updated to master

**Comment by MenacingManatee** [comment]
Oh. Apparently that changed a bunch of files, wtf. Lemme revert that lol

**Comment by MenacingManatee** [comment]
Much better. Apparently the pull failed partway through since I had a daemon running in the background

File: config/maps.txt [hunk 1]
```diff
@@ -17,6 +17,9 @@ map yogstation
 	votable
 endmap
 
+map asteroidstation
+endmap
```
Comment (JamieD1) [inline config/maps.txt:23]:
```suggestion
	maxplayers 40
endmap
```

Comment (MenacingManatee) [reply config/maps.txt:23]: I'll probably want this at minplayers 35 once testing is done, but for now using it for low-midpop makes sense

File: _maps/asteroidstation.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+	"map_name": "AsteroidStation",
```
Comment (JamieD1) [inline _maps/asteroidstation.json]:
```suggestion
	"map_name": "[ALPHA] AsteroidStation",
```

**Pull request merged**